src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -421,7 +421,7 @@ Object checkReceiver(Object recv) {
                                                recv.getClass().getName(), caller.getName());
                     throw new IncompatibleClassChangeError(msg);
                 } else {
-                    throw new NullPointerException();
+                    throw new NullPointerException("Cannot invoke " + member + " with null receiver");
```

Comment:
```suggestion
                    String msg = String.format("Cannot invoke %s with null receiver",
                                               member);
                    throw new NullPointerException(msg);
```

Reply:
Good catch.   Will update it.

Reply:
I now reminded myself that java.base is built with -XDstringConcat=inline to avoid the bootstrap issue.   In any case, I updated the patch to use `String.format` to be consistent with the ICCE message formatting.